src/app/components/services/study-list/study-list.service.js

```diff
@@ -5,183 +5,170 @@
+            /**
+             * Perform both partial and case insensitive matching
+             * for string values and study object keys
+             */
+            _.forOwn(obj, function (v, k) { // iterate through object's properties
+                var pair = [];
+
+                // Only add string values of object's key and property to the search pair
+                if (typeof v === 'string') {
+                    pair.push(v);
+                }
+                if (typeof k === 'string') {
+                    pair.push(k);
+                }
+
+                _.each(pair, function (searchString) {
+                    var idx = hasKeywordByIndex(searchString, searchKeywords);
+                    if (idx >= 0) {
+                        keyMap.push(searchKeywords[idx]);
+                        isFound = true;
+                    }
+                });
 
-            _.each(pair, function (searchString) {
-                var idx = hasKeywordByIndex(searchString, searchKeywords);
-                if (idx >= 0) {
-                    keyMap.push(searchKeywords[idx]);
-                    isFound = true;
+                if (isFound && op === 'OR') {
+                    return false;
                 }
             });
 
-            if (isFound && op === 'OR') {
-                return false;
+            if (isFound && op === 'AND') {
+                return _.uniq(keyMap).length == searchKeywords.length;
             }
-        });
-
-        if (isFound && op === 'AND') {
-            return _.uniq(keyMap).length == searchKeywords.length;
-        }
-
-        if (isFound && op === 'OR') {
-            return true;
-        }
-
-        return false;
-    };
-
-    /**
-     * Find the argumented string in one of the keywords
-     * and return the index of first matched keyword element.
-     * @memberof StudyListService
-     * @param str - String to be searched for
-     * @param keywords - Array of keywords to search in
-     * @returns index {number}
-     */
-    var hasKeywordByIndex = function (str, keywords) {
-        return _.findIndex(keywords, function (searchKeyword) {
-            return str.match(new RegExp(searchKeyword, 'i'));
-        });
-    };
-    /**
-     * Filter studies by search keywords
-     * @memberof StudyListService
-     * @param searchKeywords
-     * @returns {Array} - studies which contains search keywords
-     */
-    service.showStudiesByKeys = function (searchKeywords, operator) {
-        _.forEach(this.studyList, function (s) {
-            _.forEach(collectSearchTargetObjects(s), function (_obj) {       // Iterate through searching objects of a study
-                s.hide = !containsSearchKeys(_obj, searchKeywords, operator);  // Hide the study when it does not contain search keywords
-                if (!s.hide) return false;                                     // Exit loop when it contains search keywords
+
+            return !!(isFound && op === 'OR');
```

Comment:
Double negation should not be necessary as both values returned are already boolean.